test/distributed/test_multi_threaded_pg.py

```diff
@@ -127,6 +127,13 @@ def test_all_reduce(self):
         with self.assertRaisesRegex(NotImplementedError, "only supports SUM on threaded pg for now"):
             dist.all_reduce(output, op=ReduceOp.MAX)
 
+    def test_random_seed_consistency(self):
+        rng = torch.Generator()  # TODO: we need to hide this instantiation from test
+        self_tensor = torch.rand(3, 3, generator=rng)  # TODO: we need to hide generator=rng from test
+        rank_0_tensor = self_tensor.clone()
```

Comment:
This is just to test whether rng works in multithreaded or not? That should already been tested by rng tests itself, we don't need to add a test like this.

Reply:
That's right. The goal I want to achieve is hiding line 131 and the explicit `generator=rng` in line 132 (otherwise we have to write this explicitly in every `rand()` call in tests). Currently this is the blocker. I'm digging a good solution. Any idea on it?

Reply:
Wouldn't just use `torch.set_rng_state` should work for this?

Reply:
If I understand correctly, `torch.set_rng.state` copies the passed state into `torch.default_generator` as its new internal state. However, this doesn't make each thread use its own RNG. The RNG they use is still the shared one. Right?